test/unit/decode_fixtures.test.cpp

```diff
@@ -186,9 +186,169 @@ TEST_CASE("Read feature multipolygon")
     CHECK(mpoly[0][0][4].y == 0);
 }
 
-TEST_CASE("Read invalid: missing geometry type field")
+TEST_CASE("Read: UNKNOWN geometry type, results in no features")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/016/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    CHECK(fm.empty());
+}
+
+TEST_CASE("Read: missing geometry type field, results in feature collection and no layer")
 {
     std::string buffer = open_tile("test/mvt-fixtures/fixtures/003/tile.mvt");
     auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
     CHECK(fm.empty());
 }
+
+TEST_CASE("Read: missing layer extent is assigned the default value")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/009/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    REQUIRE(f.geometry.is<mapbox::geometry::point<std::int64_t>>());
+    auto pt = f.geometry.get<mapbox::geometry::point<std::int64_t>>();
+    // just creating a point is a check that we are handling with a default extent,
+    // but we'll want to assert on extent somewhere in the future
+}
+
+TEST_CASE("Read: A Layer value property is listed as 'string' but encoded as std::int64_t, results in a skip and no properties on feature")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/010/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    CHECK(f.properties.empty());
+}
+
+TEST_CASE("Read: two layers with the same name value, but only the first layer added is kept")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/015/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    // right now we default to the std::map::emplace functionality, which will automatically
+    // remove an element where the key currently exists - meaning the first layer added
+    // will be the final layer. Is this what we want? Can we override the feature
+    // collection map to throw an exception? Something to benchmark and think about.
```

Comment:
@flippmoke @GretaCB some questions ☝️ for how we should handle duplicate layer names

Reply:
We need to have a strict guarantee that duplicate layer names will not exist in many different levels of processing. Therefore, I think this is the correct step to follow in this case.

Reply:
Great, I'll work on an update/release of mvt-fixtures to cover this.

Reply:
mvt-fixtures updated in https://github.com/mapbox/vector-tile/pull/28/commits/e89e644fbf0932487165f1edfb56b8f3bd3f46ea

Reply:
@flippmoke just to clarify do you think we should keep it as-is or throw an exception?

Reply:
Keep it as is, we don't really want an exception -- just ignoring one seems like the proper result!

Reply:
Great! I will do nothing! 😄